ikarth/wfc_2019f
# Rework project

Hi,

Thanks for your work of this implementation.

I was playing around this morning. And i saw the tests was mixed with the source code.
Step by step i have reworked and cleaned up the full project. Hope it is not too much.

- Use pytest for tests (tests are now in `tests` directory)
    - you can type `pytest` to run everything (you also can select a subset with command line argument)
- Use `black` to auto format the source code
    - There was a lot of mismatches of indentation, too long to fix manually
    - It looks a huge change, but it is only indentation changes
- Fix tests and impl (there was small inconsistencies)
- Use sphinx to generate the graphvis files (check the `README.md` for usage). Graphvis files are now located at `doc/dot/`
- Update python project files `pyproject.toml`, `setup.py`

I did not touch anything of the code implementation itself. That's only about project quality.

If you don't like it, please at least check `114db0b1a0706aa4412301d1c66822cd73637cf7`, there is few typo you should fix.

Regards,
